pkg/fetch/version_fetcher.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package fetch
+
+import (
+	"fmt"
+
+	"github.com/k14s/semver/v4"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/clusterclient"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// VersionFetcher supplies a generic interface for retrieving versions for components of the system
+type VersionFetcher interface {
```

Comment:
I could not find a place where this interface was used.
According to go's best practices the interfaces should be defined near the place they are used. Which will allow you to have smaller interfaces